### ISSUE TITLE
thoughts on onnx data type INT64 support? 

### ISSUE THREAD

**Issue by qigtang** [onnx/onnx-tensorrt#5]
I am trying to deploy onnx model using tensorrt, but cannot find tensorrt support for int64 data type. 

nvinfer1::ITensor supprts the following types

kFLOAT 	FP32 format.
kHALF 	FP16 format.
kINT8 	INT8 format.
kINT32 	INT32 format.

In the onnx model exported from pytorch, I have several usages of int64
1. Shape operator generate int64 type tensor per definition
2. some constants are int64 type.

Any thoughts on how to address that? Should we modify pytorch export  to avoid int64? Assume
we never use a number > 2G? Or should we add that into onnx-tensorrt ?

**Comment by yinghai** [comment]
If your data fit into int32, I think you can just convert them to int64, especially for shapes. :| I think it didn't make sense to export dimensions as int64 in the first place...

**Comment by RizhaoCai** [comment]
I am very very puzzled. Seems we cannot avoid using int64 in PyTorch code, like shape operation, reshape (view) operation.

**Comment by xiaoxiaotao** [comment]
You solve the question?when inference faceboxes of pytorch with onnx-tensorrt ,I have the same question.
error:
Input filename:   /home/tao/useful/faceboxes_tensorrt_cuda/src/faceboxes_v1.onnx
ONNX IR version:  0.0.3
Opset version:    9
Producer name:    pytorch
Producer version: 0.4
Domain:           
Model version:    0
Doc string:       
----------------------------------------------------------------
Unsupported ONNX data type: INT64 (7)
ERROR: Parameter check failed at: ../builder/Layers.cpp::ConstantLayer::1585, condition: weights.type == DataType::kFLOAT || weights.type == DataType::kHALF || weights.type == DataType::kINT32
While parsing node number 100 [Gather]:
ERROR: onnx2trt_utils.hpp:275 In function convert_axis:
[8] Assertion failed: axis >= 0 && axis < nbDims
ERROR: failed to parse onnx file

**Comment by kealennieh** [comment]
any updates about this data type problem ? Thanks.

**Comment by yanhn** [comment]
if you have some ops like this:
`o.view(o.size(0), -1)` please change to `o.view(int(o.size(0)), -1)`. It worked for me.

**Comment by xiaoxiaotao** [comment]
@yanhn thank you ,your solution works.

**Comment by BloomBerry** [comment]
@yanhn Thank you so much, It worked for me too!

**Comment by kevinch-nv** [comment]
TensorRT currently doesn't have plans to support INT64 types. In later TRT versions, (>=7.0) we downcast INT64 to INT32 in TensorRT, as most frameworks use INT64 type for indexing when converting to ONNX.

Closing this issue.

**Comment by zhyncs** [comment]
> TensorRT currently doesn't have plans to support INT64 types. In later TRT versions, (>=7.0) we downcast INT64 to INT32 in TensorRT, as most frameworks use INT64 type for indexing when converting to ONNX.
> 
> Closing this issue.

Hi @kevinch-nv  we are recently trying out TRT 8.6.1.6, using TensorFlow tf2trt version 1.15 while porting the latest TensorFlow tf2trt master code. On our recommendation system rank CTR model, we are using tf2trt to optimize the model and found that the log shows `Unsupported data type int64`. Would like to know if this is as expected, and if it conflicts with your statement in the issue reply that `In later TRT versions, (>=7.0) we downcast INT64 to INT32 in TensorRT`? Thank you!

**Comment by MickaMickaMicka** [comment]
> TensorRT currently doesn't have plans to support INT64 types. In later TRT versions, (>=7.0) we downcast INT64 to INT32 in TensorRT, as most frameworks use INT64 type for indexing when converting to ONNX.
> 
> Closing this issue.

Conversion works for me, but intialization time is very high. For a yolo v5 nano it takes 1-2 minutes (probably because of the conversion step?) on an edge device, which is way too long. Is there  a way to save the converted model to file once and use that model in the future?
